### PR TITLE
course: task 2.5 decimal validator

### PR DESCRIPTION
**Pull request by BorislavAnchev**
The decimal validator task.

### CODE REVIEW

File: 25DecimalValidator/DecimalValidator.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const validate = () => {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.js]: A better design would be validation to be separated from the UI as it will make it re-usable.

**Comment by BorislavAnchev** [comment]
Fixes are ready.

File: 25DecimalValidator/DecimalValidator.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const validate = (number) => {
+    if (/^[-+]?[0-9]{1,10}([.,][0-9]{1,5})?$/.test(number)) {
+        alert("The number is OK!");
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.js]: alert is part of the UI.

File: 25DecimalValidator/DecimalValidator.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const validate = (number) => {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.js]: Maybe:
const isDecimalNumber = (value) => {
}

sounds better.

**Comment by BorislavAnchev** [comment]
Changes are made. I also removed the ES6 syntax, based on the reviews of the previous task.

File: 25DecimalValidator/DecimalValidator.js (hunk 3 above)
Comment (BorislavAnchev) [reply 25DecimalValidator/DecimalValidator.js]: Done.

File: 25DecimalValidator/DecimalValidator.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <form name="decimalValidatorForm">
+        <input type="text" name="inputField"><br>
+        <input type="button" name="check_btn" class="button" onclick="emmitAlert(document.decimalValidatorForm.inputField.value)" value="Check"> 
+      </form>
+    </div>
+  </body>
+
+  <script>
+    const emmitAlert = function(number) {
+      if(validate(number)) {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.html]: still used as validate.

File: 25DecimalValidator/DecimalValidator.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const isDecimalNumber = function(number) {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.js]: const is also part of ES6 ...

Comment (BorislavAnchev) [reply 25DecimalValidator/DecimalValidator.js]: const changed to var.

File: 25DecimalValidator/DecimalValidator.html (hunk 4 above)
Comment (BorislavAnchev) [reply 25DecimalValidator/DecimalValidator.html]: The name is corrected.

**Comment by BorislavAnchev** [comment]
New fixes ready for review.

File: 25DecimalValidator/DecimalValidator.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="main">
+      <form name="decimalValidatorForm">
+        <input type="text" name="inputField"><br>
+        <input type="button" name="check_btn" class="button" onclick="emmitAlert(document.decimalValidatorForm.inputField.value)" value="Check"> 
+      </form>
+    </div>
+  </body>
+
+  <script>
+    var emmitAlert = function(number) {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.html]: sounds like is checksEnteredValue instead of emitting alert. The emitting is a result of the check.

Comment (BorislavAnchev) [reply 25DecimalValidator/DecimalValidator.html]: Changed.

**Comment by BorislavAnchev** [comment]
Fixes ready for review.

File: 25DecimalValidator/DecimalValidator.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+var isDecimalNumber = function(number) {
+  if (/^[-+]?[0-9]{1,10}([.,][0-9]{1,5})?$/.test(number)) {
```
Comment (mgenov) [inline 25DecimalValidator/DecimalValidator.js]: could be just: return /^[-+]?[0-9]{1,10}([.,][0-9]{1,5})?$/.test(number)

Comment (BorislavAnchev) [reply 25DecimalValidator/DecimalValidator.js]: Done.

**Comment by BorislavAnchev** [comment]
Fixes ready for review.

**Pull request merged**